src/plugins/data_source_management/public/components/data_source_view/data_source_view_error.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@osd/i18n';
+import { ApplicationStart } from 'opensearch-dashboards/public';
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiPopover,
+  EuiPopoverFooter,
+  EuiText,
+} from '@elastic/eui';
+import { ErrorIcon } from '../custom_database_icon';
+import { DataSourceDropDownHeader } from '../drop_down_header';
+import { switchDefaultButton } from '../toast_button/switch_default_button';
+
+interface DataSourceViewErrorProps {
+  application?: ApplicationStart;
+  dataSourceId: string;
+  showSwitchButton: boolean;
+  handleSwitchDefaultDatasource?: () => void;
+}
+
+export const DataSourceViewError = ({
+  application,
+  dataSourceId,
+  showSwitchButton,
+  handleSwitchDefaultDatasource,
+}: DataSourceViewErrorProps) => {
+  const [showPopover, setShowPopover] = useState<boolean>(false);
+
+  const switchButton = (
+    <EuiButton
+      data-test-subj="dataSourceViewErrorSwitchButton"
+      fill={false}
+      size="s"
+      onClick={handleSwitchDefaultDatasource}
+    >
+      {i18n.translate('dataSourcesManagement.dataSourceViewError.switchToDefaultDataSource', {
+        defaultMessage: 'Switch to default data source',
+      })}
+    </EuiButton>
+  );
+
+  const iconButton = (
+    <EuiButtonIcon
+      className="euiHeaderLink"
+      data-test-subj="dataSourceViewErrorHeaderLink"
+      aria-label={i18n.translate('dataSourceError.dataSourceErrorHeaderLink', {
+        defaultMessage: 'dataSourceErrorMenuHeaderLink',
```

Comment:
can we have more consistent names, className, data-test-subj, aria-label are all different